src/main/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsv.java

```diff
@@ -155,26 +156,48 @@ private static void exportFieldValues(final PDFInteractiveForm form, final CSVPr
         while (fieldIterator.hasNext()) {
             final PDFField field = fieldIterator.next();
 
-            final String formattedValue = field.getFormattedValue();
-            if (formattedValue == null) {
-                final List value = field.getValueList();
-                if (value == null) {
-                    printer.print("");
-                    if (LOGGER.isLoggable(Level.WARNING)) {
-                        LOGGER.warning(field.getQualifiedName() + " has no value!");
-                    }
+            List value;
+            if (VersionUtils.pdfjtIsBeforeVersion4()) {
+                value = field.getValueList();
+                validateField(printer, field, value);
+            } else {
+                final String formattedValue = field.getFormattedValue();
```

Comment:
Since the _getFormattedValue()_ API did not exist in PDF Java Toolkit 4.0.0 (it didn't show up until 4.7.0), it would really be nice to check the version number more accurately in this case.